e2e/cypress/integration/ui/infra-proxy/chef-servers.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('chef server', () => {
+  const now = Cypress.moment().format('MMDDYYhhmm');
+  const cypressPrefix = 'infra';
+  const serverName = `${cypressPrefix} server ${now}`;
+  const generatedServerID = serverName.split(' ').join('-');
+  const customServerID = `${cypressPrefix}-custom-id-${now}`;
+  const serverFQDN = 'chef-server-1617089723092818000.com';
+  const serverIP = '176.119.50.159';
+
+  before(() => {
+    cy.adminLogin('/infrastructure/chef-servers').then(() => {
+      const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
+      cy.get('app-welcome-modal').invoke('hide');
+    });
+    cy.restoreStorage();
+  });
+
+  beforeEach(() => {
+    cy.restoreStorage();
+  });
+
+  afterEach(() => {
+    cy.saveStorage();
+  });
+
+  describe('chef server list page', () => {
+
+    it('can add a server', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+      cy.get('[data-cy=add-name]').type(serverName);
+      cy.get('[data-cy=id-label]').contains(generatedServerID);
+      cy.get('[data-cy=add-fqdn]').type(serverFQDN);
+      cy.get('[data-cy=add-ip-address]').type(serverIP);
+
+      cy.get('[data-cy=add-button]').click();
+      cy.get('app-chef-servers-list chef-modal').should('not.be.visible');
+
+      // verify success notification and then dismiss it
+      // so it doesn't get in the way of subsequent interactions
+      cy.get('app-notification.info').should('be.visible');
+      cy.get('app-notification.info chef-icon').click();
+
+      cy.contains(serverName).should('exist');
```

Comment:
This seems too weak--not even clear what the scope is for the `contains`. What you have on L!23-124 is much tighter:
```suggestion
      cy.get('app-chef-servers-list chef-tbody chef-td')
          .contains(serverName).should('exist');
```